src/olympia/devhub/tasks.py

```diff
@@ -80,8 +80,9 @@ def submit_file(addon_pk, upload_pk):
 @atomic
 def create_version_for_upload(addon, upload):
     if (addon.fileupload_set.filter(created__gt=upload.created,
-                                    version=upload.version).exists()
-            or addon.versions.filter(version=upload.version).exists()):
+                                    version=upload.version).exists() or
+            Version.unfiltered.filter(addon=addon,
+                                      version=upload.version).exists()):
```

Comment:
Can we untangle this if statement before it get's even bigger? Again, as proposed earlier something like

```python
fileupload_exists = addon.fileupload_set.filter(...).exists()
version_exists = Version.unfiltered.filter(...).exists()

if fileupload_exists or version_exists:
    log.info(...)
```